### DISCUSSION TITLE
PASSWORD_RESET_URL_ALLOW_LIST with includes a # generates an invalid link in the email

### DISCUSSION THREAD

**Discussion by ludi81** [directus/directus#13086] [Archive]
### Describe the Bug

Hi,

we use a # in our link for password rest. Therefore we set the PASSWORD_RESET_URL_ALLOW_LIST as follows:
`PASSWORD_RESET_URL_ALLOW_LIST='https://URL/app/#/password/reset/'`

The system generates the following url
`https://URL/app?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InNvZmlhQGx1ZGVzY2hlci5hdCIsInNjb3BlIjoicGFzc3dvcmQtcmVzZXQiLCJoYXNoIjoiN2Q2NmEyOGYiLCJpYXQiOjE2NTE0NzU1NDgsImV4cCI6MTY1MTU2MTk0OCwiaXNzIjoiZGlyZWN0dXMifQ.8LC5tbTGBcmSanifhrx_8nkkUzOEnRdYbYEC_wHRHEI#/password/reset`

instead of
`https://URL/app/#/password/reset/?token=....`

The token will be inserted in before the # in the provided url.
The problem is in the requestPasswordReset function of the node_modules/directus/dist/services/users.js file

**Comment by rijkvanzanten** [comment]
This is correct. Query parameters come before fragments in URLs, as per RFC 3986:

```
A relative reference takes advantage of the hierarchical syntax
   ([Section 1.2.3](https://www.rfc-editor.org/rfc/rfc3986#section-1.2.3)) to express a URI reference relative to the name space
   of another hierarchical URI.

      relative-ref  = relative-part [ "?" query ] [ "#" fragment ]
```

https://www.rfc-editor.org/rfc/rfc3986#section-4.1

**Reply by informatikTirol** [reply]
Angular has it the other way around when using 
`useHash=true `
[https://angular.io/api/router/ExtraOptions#useHash](https://angular.io/api/router/ExtraOptions#useHash)
in the router module. 

The routing breaks if you have the query parameter before the fragment.

**Reply by ludi81** [reply]
I am using angular as well. I needed to change the source of the core to fix the problem
